apps/frontend/src/utils/useHttp.ts

```diff
@@ -0,0 +1,36 @@
+import axios, { type AxiosRequestConfig } from 'axios';
+import type { Dispatch, AnyAction } from '@reduxjs/toolkit';
+
+import { logout } from '@/store/actions/auth';
+
+const useHttp = async (
+	axiosParams: AxiosRequestConfig,
+	dispatch: Dispatch<AnyAction>,
```

Comment:
Maybe we can take all the actions you have and pass them inside here as the generic instead of `AnyAction` if you want to do that we can try together.